assets/app/scripts/services/applicationGenerator.js

```diff
@@ -59,13 +34,32 @@ angular.module("openshiftConsole")
               protocol: parts[1].toUpperCase()
             });
         });
+
+        // Since the exposed ports in Docker image metadata are not in any
+        // order, sort the ports from lowest to highest.
+        ports.sort(function(left, right) {
+          return left.containerPort - right.containerPort;
+        });
+
         return ports;
       };
-      var ports = input.image.dockerImageMetadata.Config ? parsePorts(input.image.dockerImageMetadata.Config.ExposedPorts) : [];
-      if(ports.length === 0 && input.image.dockerImageMetadata.ContainerConfig){
-        ports = parsePorts(input.image.dockerImageMetadata.ContainerConfig.ExposedPorts);
+
+      var ports = image.dockerImageMetadata.Config ? parsePortsFromSpec(image.dockerImageMetadata.Config.ExposedPorts) : [];
```

Comment:
would it be better to do `$parse('dockerImageMetadata.Config.ExposedPorts')(image) || []`?